django/db/models/sql/query.py

```diff
@@ -1030,19 +1032,22 @@ def resolve_expression(self, query, *args, **kwargs):
         for key, value in clone.annotations.items():
             resolved = value.resolve_expression(query, *args, **kwargs)
             if hasattr(resolved, 'external_aliases'):
-                resolved.external_aliases.update(clone.alias_map)
+                resolved.external_aliases.update(clone.external_aliases)
```

Comment:
I didn't play with this part a lot but this seemed like an error given we check for `hasattr(resolved, 'external_aliases')`. It also makes the update easier.